main.py

```diff
@@ -117,32 +120,45 @@ def _restore_status(self, switches_status, interfaces_status):
                 error = ('Error while restoring interface status. The '
                          f'interface {interface_id} does not exist.')
                 raise KeyError(error)
-
+        # links
+        for link, state, in self.links_state.items():
+            try:
+                if state:
+                    self.links[link].enable()
+                else:
+                    self.links[link].disable()
+            except KeyError:
+                error = ('Error restoring link status.'
+                         f'The link {link} does not exist.')
+                raise KeyError(error)
         log.info('Network status restored.')
```

Comment:
This logic is not working on the first run, because this code snippet requires the `add_links` method to fill in the self.links. Due to this problem, it was moved to the draft again.